cmd/katana/main.go

```diff
@@ -207,23 +186,43 @@ func init() {
 
 func defaultResumeFilename() string {
 	homedir, err := os.UserHomeDir()
-	if err != nil {
-		gologger.Fatal().Msgf("could not get home directory: %s", err)
-	}
+	handleError("could not get home directory", err)
 	configDir := filepath.Join(homedir, ".config", "katana")
 	return filepath.Join(configDir, fmt.Sprintf("resume-%s.cfg", xid.New().String()))
 }
 
-// cleanupOldResumeFiles cleans up resume files older than 10 days.
+func setupCloseHandler(runner *runner.Runner, resumeFilename string) {
+	go func() {
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+		for range c {
+			gologger.DefaultLogger.Info().Msg("- Ctrl+C pressed in Terminal")
+			runner.Close()
+
+			gologger.Info().Msgf("Creating resume file: %s\n", resumeFilename)
+			err := runner.SaveState(resumeFilename)
+			if err != nil {
+				gologger.Error().Msgf("Couldn't create resume file: %s\n", err)
+			}
+
+			os.Exit(0)
+		}
+	}()
+}
+
 func cleanupOldResumeFiles() {
 	homedir, err := os.UserHomeDir()
-	if err != nil {
-		gologger.Fatal().Msgf("could not get home directory: %s", err)
-	}
+	handleError("could not get home directory", err)
 	root := filepath.Join(homedir, ".config", "katana")
 	filter := fileutil.FileFilters{
 		OlderThan: 24 * time.Hour * 10, // cleanup on the 10th day
 		Prefix:    "resume-",
 	}
 	_ = fileutil.DeleteFilesOlderThan(root, filter)
 }
+
+func handleError(message string, err error) {
```

Comment:
Since this is going to fail, we should be more explicit about it. Something like `failWithError` or similar name would be great